### PR TITLE
[pt2] add `SymInt` support for `linalg.vander`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #102469

### CODE REVIEW

File: aten/src/ATen/native/ReduceOps.cpp [hunk 1]
```diff
@@ -593,11 +593,11 @@ Tensor cumprod_backward(const Tensor& grad, const Tensor& input, int64_t dim, co
     cumprod(input.conj()) = cumprod(input).conj()
   */
 
-  if (input.numel() <= 1) {
+  if (input.sym_numel() <= 1) {
     return grad;
   }
   dim = at::maybe_wrap_dim(dim, input.dim());
-  const int64_t dim_size = input.sizes()[dim];
+  const int64_t dim_size = input.sym_sizes()[dim].guard_int(__FILE__, __LINE__);
```
Comment (nkaretnikov) [inline aten/src/ATen/native/ReduceOps.cpp:600]: dim_size is later used with irange and reserve.

**Review by Skylion007 — APPROVED** [review]

**Review by lezcano — APPROVED** [review]

**Pull request closed without merging**